l10n_ch_bank/views/res_bank.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_res_bank_form" model="ir.ui.view">
+        <field name="name">res.bank.form</field>
+        <field name="model">res.bank</field>
+        <field name="inherit_id" ref="base.view_res_bank_form"/>
+        <field name="arch" type="xml">
+
+            <xpath expr="//field[@name='name']" position="after">
+                <field name="country_code" invisible="True" />
+            </xpath>
+
+            <xpath expr="//field[@name='city']" position="attributes">
+                <attribute name="attrs">{'required': [('country_code', '=', 'CH')]}</attribute>
+            </xpath>
+
+            <xpath expr="//field[@name='bic']" position="attributes">
+                <attribute name="attrs">{'required': [('country_code', '=', 'CH')]}</attribute>
+            </xpath>
+
+        </field>
+    </record>
+
+</odoo>
```

Comment:
I think this should go in the `l10n_ch_base_bank` module.

The current module `l10n_ch_bank` is here for the bank data while the module `l10n_ch_base_bank` is about the changes made on the bank object.